mathesar_ui/src/component-library/tabs/TabContainer.svelte

```diff
@@ -10,23 +10,27 @@
 <script lang="ts">
   import { createEventDispatcher } from 'svelte';
   import TabComponent from './Tab.svelte';
-  import type { Tab } from './TabContainerTypes';
+  import type { Tab, TabEvents } from './TabContainerTypes';
 
-  const dispatch = createEventDispatcher();
+  const dispatch = createEventDispatcher<TabEvents>();
   const componentId = getId();
 
   export let tabs: Tab[] = [];
   export let activeTab: Tab | undefined = tabs[0];
   export let idKey = 'id';
   export let labelKey = 'label';
+  export let linkKey = 'href';
   export let allowRemoval = false;
   export let preventDefault = false;
   export let fillTabWidth = false;
   export let fillContainerHeight = false;
   export let uniformTabWidth = true;
 
   function selectActiveTab(e: Event, tab: Tab) {
-    activeTab = tab;
+    const hasLink = !!tab[linkKey];
+    if (!hasLink) {
+      activeTab = tab;
+    }
```

Comment:
Why is active tab not set here for tabs with links?

Reply:
This component is common and this property needs to be handled by the component. Binding to the property from the parent is optional, so it's not a concern if it gets changed internally in this scenario since we control the property on the parent.

Reply:
@pavish 

This is done to avoid the state mess up created by the two data binding when the user does cmd+click on the tabs with links. 

Here is a video showcasing this: https://www.loom.com/share/e2be4607c15847beac3997c5c43df58c

Reply:
@rajatvijay The solution made above is a workaround. The proper way to handle this is to select the tab only on the click event (not on the mouse down event), and to check if 'cmd/ctrl' is pressed down when click event is raised and if the tab is a link, and do nothing if it's a cmd+click and leave it to the browser to take over. I believe this is small enough to handle it in this PR itself.